File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/hardware/physical/request/AnalogDeviceRequest.java
package org.firstinspires.ftc.teamcode.utils.hardware.physical.request;

import com.michaell.looping.ScriptParameters;
import com.qualcomm.robotcore.hardware.AnalogInput;
import com.qualcomm.robotcore.hardware.HardwareMap;
import org.firstinspires.ftc.teamcode.utils.hardware.physical.inputs.AnalogData;
import org.firstinspires.ftc.teamcode.utils.hardware.physical.inputs.AnalogData;
import org.firstinspires.ftc.teamcode.utils.hardware.physical.inputs.AnalogData;

public class AnalogDeviceRequest extends ScriptParameters.Request {

    private final AnalogInput CONTROLLER;

    public AnalogDeviceRequest(String name, HardwareMap hardwareMap) {
        super(name);
        CONTROLLER = hardwareMap.get(AnalogInput.class, name);
        CONTROLLER.resetDeviceConfigurationForOpMode();
    }

    @Override
    public Object issueRequest(Object o) {
        return new AnalogData(CONTROLLER.getVoltage(), CONTROLLER.getMaxVoltage());
    }

    @Override
    public Class getOutputType() {
        return AnalogData.class;
    }

    @Override
    public Class getInputType() {
        return Object.class;
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/hardware/physical/requests/AnalogDeviceRequest.kt
package org.firstinspires.ftc.teamcode.utils.hardware.physical.requests

import com.michaell.looping.ScriptParameters
import com.qualcomm.robotcore.hardware.AnalogInput
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.utils.hardware.physical.data.AnalogData

class AnalogDeviceRequest(name: String, hardwareMap: HardwareMap) : ScriptParameters.Request(name) {
    private val CONTROLLER: AnalogInput

    init {
        CONTROLLER = hardwareMap.get(AnalogInput::class.java, name)
        CONTROLLER.resetDeviceConfigurationForOpMode()
    }

    override fun issueRequest(o: Any): Any {
        return AnalogData(CONTROLLER.voltage, CONTROLLER.maxVoltage)
    }

    override fun getOutputType(): Class<*> {
        return AnalogData::class.java
    }

    override fun getInputType(): Class<*> {
        return Any::class.java
    }
}